test/integration/targets/ansible-galaxy-collection/tasks/install.yml

```
@@ -410,6 +423,137 @@
+  copy:
+    content: |
+      collections:
+      - name: namespace7.name
+        version: "1.0.0"
+        signatures:
+          - "file://{{ gpg_homedir }}/namespace7-name-1.0.0-MANIFEST.json.asc"
+          - "{{ not_mine }}"
+          - "{{ also_not_mine }}"
+      - namespace8.name
+      - name: namespace9.name
+        signatures:
+          - "file://{{ gpg_homedir }}/namespace9-name-1.0.0-MANIFEST.json.asc"
+    dest: '{{ galaxy_dir }}/ansible_collections/requirements.yaml'
+  vars:
+    not_mine: "file://{{ gpg_homedir }}/namespace1-name1-1.0.0-MANIFEST.json.asc"
+    also_not_mine: "file://{{ gpg_homedir }}/namespace1-name1-1.0.9-MANIFEST.json.asc"
+
+- name: install collection with mutually exclusive options
+  command: ansible-galaxy collection install -r {{ req_file }} -s {{ test_name }} {{ cli_signature }}
+  vars:
+    req_file: "{{ galaxy_dir }}/ansible_collections/requirements.yaml"
+    # --signature is an ansible-galaxy collection install subcommand, but mutually exclusive with -r
+    cli_signature: "--signature file://{{ gpg_homedir }}/namespace7-name-1.0.0-MANIFEST.json.asc"
+  ignore_errors: yes
+  register: mutually_exclusive_opts
+
+- assert:
+    that:
+      - mutually_exclusive_opts is failed
+      - expected_error in actual_error
+  vars:
+    expected_error: >-
+            The --signatures option and --requirements-file are mutually exclusive.
+            Use the --signatures with positional collection_name args or provide a
+            'signatures' key for requirements in the --requirements-file.
+    actual_error: "{{ mutually_exclusive_opts.stderr }}"
+
+- name: install a collection with user-supplied signatures for verification but no keyring
+  command: ansible-galaxy collection install namespace1.name1:1.0.0 {{ cli_signature }}
+  vars:
+    cli_signature: "--signature file://{{ gpg_homedir }}/namespace1-name1-1.0.0-MANIFEST.json.asc"
+  ignore_errors: yes
+  register: required_together
+
+- assert:
+    that:
+      - required_together is failed
+      - '"ERROR! Signatures were provided to verify namespace1.name1 but no keyring was configured." in required_together.stderr'
+
+- name: install collections with ansible-galaxy install -r with invalid signatures - {{ test_name }}
+  # Note that --keyring is a valid option for 'ansible-galaxy install -r ...', not just 'ansible-galaxy collection ...'
+  command: ansible-galaxy install -r {{ req_file }} -s {{ test_name }}  --keyring {{ keyring }}
+  register: install_req
+  ignore_errors: yes
+  vars:
+    req_file: "{{ galaxy_dir }}/ansible_collections/requirements.yaml"
+    keyring: "{{ gpg_homedir }}/pubring.kbx"
+  environment:
+    ANSIBLE_COLLECTIONS_PATH: '{{ galaxy_dir }}/ansible_collections'
+
+- name: assert invalid signature is fatal with ansible-galaxy install - {{ test_name }}
+  assert:
+    that:
+    - install_req is failed
+    - '"Installing ''namespace7.name:1.0.0'' to" in install_req.stdout'
```

I suggest using `>-` to avoid having to escape things. It took me a minute to guess that `''` wasn't two quotes but one.
```suggestion
    - >-
      "Installing 'namespace7.name:1.0.0' to" in install_req.stdout
```

---

This is a pattern throughout the file already. Going to keep it like the rest for now, but as a follow up they could all be replaced to make the test suite easier to read.